### PR TITLE
Update the readme and documentation for guardian 1.0

### PR DESCRIPTION
**Pull request by hassox**
This should not be merged until Guardian 1.0 has shipped.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@ GuardianDb
 ==========
 
 GuardianDB is an extension to vanilla Guardian that tracks tokens in your
-application to prevent playback.
+applications database to prevent playback.
```
Comment (doomspork) [inline README.md]: I think you want "application's" here.

File: README.md [hunk 2]
```diff
@@ -42,100 +41,83 @@ You will then need to add a migration:
...
           timestamps()
         end
-
       end
-
     end
 ```
 
-**Do not run the migration yet.** You also need to add this to your configuration:
+**Do not run the migration yet.** You also need to finish your setup first.
```
Comment (doomspork) [inline README.md]: This doesn't read quite right, what about something like:

**Do not run the migration yet**, we need to complete our setup first:

File: README.md [hunk 3]
```diff
@@ -42,100 +41,83 @@ You will then need to add a migration:
...
+# Configuration
 
-    config :guardian_db, GuardianDb,
-           repo: MyApp.Repo,
-           schema_name: "guardian_tokens"
+```elixir
+  config :guardian_db, GuardianDb,
+         repo: MyApp.Repo,
+         schema_name: "guardian_tokens", # default
+         sweep_interval: 60 # minutes. 60 default
```
Comment (doomspork) [inline README.md]: You could combine these two fragments: `# default: 60 minutes`

File: README.md [hunk 4]
```diff
@@ -42,100 +41,83 @@ You will then need to add a migration:
...
-Now run the migration and you'll be good to go.
-
-Usage
-==========
-
-All tokens are stored in the database when initially generated.
-After that, each time they are verified, the token is looked up. If present, the
-verification continues but if it is not found, the verification is abandoned
-with an error response.
+To sweep expired tokens from your db you should add `GuardianDb` to your supervision tree.
```
Comment (doomspork) [inline README.md]: Did you mean `GuardianDb.ExpiredSweeper`?

File: README.md [hunk 5]
```diff
@@ -42,100 +41,83 @@ You will then need to add a migration:
...
+## Guardian >= 1.0
 
-```elixir
-    case Guardian.revoke! jwt, claims do
-      :ok -> # Great
-      {:error, :could_not_revoke_token} -> # Oh no GuardianDb
-      {:error, reason} -> # Oh no
-    end
-```
+GuardianDb works by hooking into the lifecycle of your token module.
```
Comment (doomspork) [inline README.md]: In other places `GuardianDb` is in backticks, did you want to do that here too?

File: README.md [hunk 6]
```diff
@@ -42,100 +41,83 @@ You will then need to add a migration:
...
 
-It's a good idea to purge out any stale tokens that have already expired.
+You'll need to add it to
```
Comment (doomspork) [inline README.md]: I think you'll want a colon here.

**Comment by hassox** [comment]
:metal: Thanks dude. Updated.

File: mix.exs [hunk 7]
```diff
@@ -28,10 +28,10 @@ defmodule GuardianDb.Mixfile do
   defp _elixirc_paths(_), do: ["lib"]
 
   defp deps do
-    [{:guardian, "~> 0.14"},
+    [{:guardian, "~> 0.14 or >= 1.0"},
```
Comment (doomspork) [inline mix.exs]: Do we need to change this to just `1.0`?

**Review by doomspork — CHANGES_REQUESTED** [review]

**Comment by hassox** [comment]
@doomspork any other changes needed on this one?

**Review by doomspork — APPROVED** [review]
You're good to go @hassox 👍

**Pull request merged**